packages/jest-cli/src/jest.js

```diff
@@ -310,10 +311,58 @@ const runCLI = (
         let testWatcher;
         let timer;
         let displayHelp = true;
+        let typeaheadLinesToClear = 0;
+
+        const typeahead = (hasteMap, patternInfo) => {
+          return new SearchSource(hasteMap, config)
```

Comment:
do we need to create a new SearchSource every time the user types? I don't know how slow it is